### PR TITLE
Add route to swap two parent structures.

### PR DESCRIPTION
**Pull request by iamanandkr**
Add route to swap two parent structures. 

## Description

Route to interact with the roo layer.

## Related Issue

[ACAS-286](https://jira.schrodinger.com/browse/ACAS-286).
[acas-roo-server](https://github.com/mcneilco/acas-roo-server/pull/342) PR.

## How Has This Been Tested?

I used curl requests to check the API,
```
curl -X POST http://localhost:3000/cmpdreg/swapParentStructures -H 'Content-Type: application/json' --data-raw '{"corpName1": "CMPD-0000001", "corpName2": "CMPD-0000002"}'
```
## TODO:
~In cases duplicate parents exist for a compound, it's still going ahead with the swapping even though duplicates should have been marked. See [acas-roo-server](https://github.com/mcneilco/acas-roo-server/pull/342) PR.~

### CODE REVIEW

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 1]
```diff
@@ -856,8 +857,31 @@ exports.updateParent = (req, resp) ->
...
+	# TODO: Do we need to keep track of who swapped the parent structure.
+	cmpdRegCall = config.all.client.service.cmpdReg.persistence.fullpath + '/parents/swapParentStructures'
+	request(
+		method: 'POST'
+		url: cmpdRegCall
+		body: req.body
+		json: true
+		timeout: 6000000
+	, (error, response, data) =>
+		if error
```
Comment (dalejerikson) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:873]: Just to keep things consistent, I would do `if !error` first followed by an `else`

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 2]
```diff
@@ -856,8 +857,31 @@ exports.updateParent = (req, resp) ->
...
+
+exports.swapParentStructures = (req, resp) ->
+	console.log "exports.swapParentStructures"
+
+	# TODO: Do we need to keep track of who swapped the parent structure.
+	cmpdRegCall = config.all.client.service.cmpdReg.persistence.fullpath + '/parents/swapParentStructures'
+	request(
+		method: 'POST'
+		url: cmpdRegCall
+		body: req.body
```
Comment (dalejerikson) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:869]: Should we do `JSON.stringify req.body`?

**Review by dalejerikson — CHANGES_REQUESTED** [review]

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 3]
```diff
@@ -856,8 +857,31 @@ exports.updateParent = (req, resp) ->
 			resp.statusCode = 500
 			resp.end "Error trying to update parent: " + error;
 	)
-	
-	
+
+exports.swapParentStructures = (req, resp) ->
+	console.log "exports.swapParentStructures"
```
[the remark this answers is not in the document]
Comment (iamanandkr) [reply modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:862]: Removed.

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 4]
```diff
@@ -856,8 +857,31 @@ exports.updateParent = (req, resp) ->
...
+
+	# TODO: Do we need to keep track of who swapped the parent structure.
```
[the remark this answers is not in the document]
Comment (iamanandkr) [reply modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:864]: Done. Thanks!

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee (hunk 2 above)
Comment (iamanandkr) [reply modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:869]: Done. Thanks!

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee (hunk 1 above)
Comment (iamanandkr) [reply modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:873]: Sure, done.

File: modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee [hunk 5]
```diff
@@ -856,8 +857,35 @@ exports.updateParent = (req, resp) ->
...
-	
+
+exports.swapParentStructures = (req, resp) ->
+	cmpdRegCall = config.all.client.service.cmpdReg.persistence.fullpath + '/parents/swapParentStructures'
+
+	username = req.session.passport?.user.username
+	if !username
+		resp.statusCode = 401
+		resp.end "No user name specified"
+		return
```
Comment (iamanandkr) [inline modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:868]: I am not sure if I need to do this since the user is authenticated.

Comment (bffrost) [reply modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee:868]: I don't think this is necessary, because the user will already be authenticated.

**Review by bffrost — APPROVED** [review]
LGTM. Feel free to remove that bit of unnecessary code, or leave it in. It won't hurt anything.

**Pull request merged**